test/harness/e2e/worker_utils.go

```diff
@@ -75,3 +77,41 @@ func GinkgoBeforeSuite() {
 		gomega.Expect(err).ToNot(gomega.HaveOccurred())
 	})
 }
+
+// copyFile copies a file from source to destination, creating the destination directory if needed.
+func CopyFile(from, to string) error {
```

Comment:
Is CopyFile from cmd/devicesimulator or internal/agent/device/fileio/writer.go can be used?